source/material_model/rheology/visco_plastic.cc

```diff
@@ -128,6 +132,19 @@ namespace aspect
           edot_ii = std::max(std::sqrt(std::fabs(second_invariant(deviator(in.strain_rate[i])))),
                              min_strain_rate);
 
+        // In case porosity lies above the melt transport threshold
```

Comment:
Eventually the documentation should be updated to indicate that the visco_plastic rheology model itself does not handle melting, but may be used by a material model that includes melt transport (or something to that effect).